metricbeat/module/elasticsearch/testing.go

```diff
@@ -0,0 +1,70 @@
+// +build !integration
+
+package elasticsearch
+
+import (
+	"io/ioutil"
+	"path/filepath"
+	"testing"
+
+	s "github.com/elastic/beats/libbeat/common/schema"
+	"github.com/elastic/beats/metricbeat/mb"
+	mbtest "github.com/elastic/beats/metricbeat/mb/testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// TestMapper tests mapping methods
+func TestMapper(t *testing.T, glob string, mapper func(mb.ReporterV2, []byte) []error) {
+	files, err := filepath.Glob(glob)
+	assert.NoError(t, err)
+	// Makes sure glob matches at least 1 file
+	assert.True(t, len(files) > 0)
+
+	for _, f := range files {
+		t.Run(f, func(t *testing.T) {
+			input, err := ioutil.ReadFile(f)
+			assert.NoError(t, err)
+
+			reporter := &mbtest.CapturingReporterV2{}
+			errors := mapper(reporter, input)
```

Comment:
I think we could also obtain the errors with `reporter.GetErrors()` and then we wouldn't need to also return them.

Reply:
See comment above, it's not the same errors.